### PR TITLE
HomeWork1/2/3

### PR DESCRIPTION
**Pull request by Java-Dev-Art**

### CODE REVIEW

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+        int m = (s - sec) / 60; //min
+        int min = m % 60;
+        double h = (m - min) / 60;//hour
+        double hour = h % 60;
+        double d = (h - hour) / 24;
+        double day = d % 24;
+        double w = (d - day) / 7;
+//        double week = w % 7;
+
+        System.out.println("A :" + m + " минут, " + s + " секунд");
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: if we get amount of minutes, amount of seconds changes. Also if we get amount of hours, amount of minutes should change. Fix it

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+        System.out.println("Стало :" + reversNum);
+        System.out.println("");
+//        Task #5 ******************************************************************************************************
+//        Дано любое натуральное 4-х значное число. Верно ли, что все цифры числа
+//          различны?
+        System.out.println("Task #5");
+        int natNumber = 5786;
+
+        char[] chars = String.valueOf(natNumber).toCharArray();
+        if ((chars[0] != chars[1]) && (chars[0] != chars[2]) && (chars[0] != chars[3]) && (chars[1] != chars[2]) && (chars[1] != chars[3])) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: char[2] != char[3]

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+        * Task #6
+        *
+            Задание 6
+            Создайте число. Определите, является ли число трехзначным. Определите, является
+            ли его последняя цифра семеркой. Определите, является ли число четным.
+        * */
+        System.out.println("Task #6");
+        int numberT = 234;
+        char[] charsNumberT = String.valueOf(numberT).toCharArray();
+        if (charsNumberT.length < 3) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: if charsNumberT.length = 4?

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+            System.out.println("Целое число " + numberT + " не трехзначное");
+        } else {
+            System.out.println("Целое число " + numberT + " трехзначное");
+            if (charsNumberT[2] == 7) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: better not use determined values (charsNumberT[2]), these must be variables

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+                System.out.println("Последняя цифра равна 7.");
+            } else {
+                System.out.println("Последняя цифра не равна 7.");
+            }
+        }
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: checking if number is even

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+        * Task #7
+        * Имеется прямоугольное отверстие размерами a и b, определить, можно ли его
+          полностью закрыть круглой картонкой радиусом r.
+        * */
+        System.out.println("Task #7");
+        int sideA = 12;
+        int sideB = 3;
+        int radius = 45;
+//        int radiusAb = (sideA * sideA + sideB * sideA) / 4; //45
+//        System.out.println(radiusAb);
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: delete unnecessary comment

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+        System.out.println("");
+        /*
+        * Task #9
+        *
+            Задание 9
+            Изменить пример с суммой чисел таким образом, чтобы рассчитывалась не сумма, а
+            произведение, т.е. факториал числа.
+        * */
+        System.out.println("Task #9");
+//?????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: delete unnecessary comment

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+          нему слово «рублей» в правильном падеже.
+        * */
+        System.out.println("Task #8");
+        Random random = new Random();
+        int x = random.nextInt(28);
+        if (x == 1) {
+            System.out.println(x + " рубль");
+        } else if (x > 1 && x <= 4) {
+            System.out.println(x + " рубля");
+        } else if (x > 4 && x < 28) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: for example, if x = 21?

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+            Изменить пример с суммой чисел таким образом, чтобы рассчитывалась не сумма, а
+            произведение, т.е. факториал числа.
+        * */
+        System.out.println("Task #9");
+//?????????????????????????????????????????????????????????????????????????????????????????????????????????????????????
+        double sumNumbers = 1;
+        int randomNumbers = (int) (Math.random() * 20);
+        int count = 0;
+
+        while (randomNumbers != 0) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: it is not factorial. we get number and need in result 1*2*...*randomNumbers. p.s. for a single value, use a single number in the name (randomNumber, not randomNumbers)

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+            count++;
+            randomNumbers = (int) (Math.random() * 20);
+            if (count != 0) {
+                System.out.println(count + ": " + sumNumbers);
+            } else {
+                System.out.println("Косяк на касяке");
+            }
+        }
+
+//
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: unnecessary spaces

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+        /*
+        * Task #11
+        *Имеется целое число, определить является ли это число простым, т.е.
+        делится без остатка только на 1 и себя.
+        * */
+        System.out.println("Task #11");
+
+        int naturalNumber = (int) (Math.random() * 45);
+
+        if ((naturalNumber % naturalNumber) == 0 && (naturalNumber % 1) == 0) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: wrong condition

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+        }
+        System.out.println("");
+        /*
+         * Task #12
+         * Найдите сумму первых n целых чисел, которые делятся на 3.
+         * */
+        System.out.println("Task #12");
+        int integerNumbers = (int) (Math.random() * 32);
+        int sumIntegerNumbers = 0;
+        for (int i = 1; i < 10; i++) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: why i <10?

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 13]
```diff
@@ -0,0 +1,355 @@
...
+            if ((integerNumbers % 3) == 0) {
+                sumIntegerNumbers += integerNumbers;
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: we several times sum one number, but need get n different numbers and sum them

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
+
+            }
+
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: unnecessary spaces, check in all files

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 15]
```diff
@@ -0,0 +1,355 @@
...
+        /*
+        * Task #13
+        * Создать последовательность случайных чисел, найти и вывести наибольшее
+        из них.
+        * */
+        System.out.println("Task #13");
+        int[] randomNumber = new int[16];
+        int maxValueNumber = 0;
+
+        for (int index = 0; index < randomNumber.length - 1; index++) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: if we check < randomNumber.length - 1, we don't check last number, stop on pre-last

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 16]
```diff
@@ -0,0 +1,355 @@
...
+        * Task #14
+        *Создать массив оценок произвольной длины, вывести максимальную и
+        минимальную оценку, её (их) номера.
+        *
+        * */
+        System.out.println("Task #14");
+        int[] rating = new int[10];
+        int maxRating = rating[0];
+        int minRating = rating[0];
+        for (int index = 0; index < rating.length - 1; index++) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: if we check < rating.length - 1, we don't check last number

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 17]
```diff
@@ -0,0 +1,355 @@
...
+        System.out.println("Task #15");
+
+        int[] randomMass = new int[7];
+        for (int index = 0; index < randomMass.length - 1; index++) {
+            randomMass[index] = (int) (Math.random() * 5);
+        }
+        System.out.println("Массив до сортировки :" + Arrays.toString(randomMass));
+        for (int start = 0; start < randomMass.length - 1; start++) {
+            for (int index = 0; index < randomMass.length - 1 - start; index++) {
+                if (randomMass[index] > randomMass[index + 1]) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: we need reverse array

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 18]
```diff
@@ -0,0 +1,355 @@
...
+        System.out.println(Arrays.toString(randomValue));
+        for (int index = 0; index < randomValue.length - 1; index++) {
+            if (randomValue[index] > maxRandomValue) {
+                maxRandomValue = randomValue[index];
+
+            }
+            if (randomValue[index] < minRandomValue) {
+                minRandomValue = randomValue[index];
+            }
+            for (int indexRandom = minRandomValue; indexRandom < randomValue.length - 1; indexRandom++) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: indexRandom < maxRandomValue

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java [hunk 19]
```diff
@@ -0,0 +1,355 @@
...
+        System.out.println("Task #17");
+        int[][] twoDim = new int[7][7];
+
+        System.out.println(twoDim.length);
+        for (int indexI = 0; indexI < twoDim.length; indexI++) {
+            for (int indexJ = 0; indexJ < twoDim[indexI].length; indexJ++) {
+                if (indexI == twoDim.length - 1 || indexI == indexJ + 1) {
+                    twoDim[indexI][indexJ] = 1;
+                }
+                if (indexJ == indexI || indexJ == twoDim.length - 1 - indexI) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: you can union two condition for value = 1

File: src/HomeWork3/HomeWork3.java [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+7)Напишите программу, определяющую сумму всех нечетных чисел от 1 до 99
+
+1-я методичка - до конца со всеми Задачами.
+краткая инфа по занятию так же есть в выдержке Lesson3 (папка presentations)
+*/
+//        Task #1 ******************************************************************************************************
+        System.out.println("task #1");
+        int km = 10;
+        for (int i = 1; i <= 7; i++) {
+            km += (km * 0.1);
```
Comment (svetamolchanova) [inline src/HomeWork3/HomeWork3.java]: if we make km int, we get round value

File: src/HomeWork3/HomeWork3.java [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+        for (int i = 3; i <= 24; i += 3) {
+            amoebas *= 2;
+            System.out.println("Прошло\t" + i + " ч. Количкство амеб : " + amoebas);
+        }
+        System.out.println("");
+//        Task #3*******************************************************************************************************
+        System.out.println("Task #3");
+        int sum = 0;
+//      Тут я не уверен
+        for (int i = 1; i < 256; i++) {
```
Comment (svetamolchanova) [inline src/HomeWork3/HomeWork3.java]: inclusive 256

File: src/HomeWork3/HomeWork3.java [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+            sum += i;
+            System.out.println(i + " сумма = " + sum);
+
+        }
+        System.out.println("");
+//        Task #4 ******************************************************************************************************
+        System.out.println("Task #4");
+
+        int a = 3;
+        int b = 4;
```
Comment (svetamolchanova) [inline src/HomeWork3/HomeWork3.java]: if we write negative number?

File: src/HomeWork3/HomeWork3.java [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+        System.out.println("");
+//        Task #7 ******************************************************************************************************
+        System.out.println("Task #7");
+        int sum2 = 0;
+        int[] numL = new int[100];
+        for (int q = 1; q < numL.length; q++) {
+            numL[q] = q;
+            if ((numL[q] % 2) == 0) {
+
+            } else {
```
Comment (svetamolchanova) [inline src/HomeWork3/HomeWork3.java]: better use contion if ((numL[q] % 2) != 0)

File: src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java (hunk 1 above)
Comment (Java-Dev-Art) [reply src/HomeWork3/FirstTrainingManualToEnd/FirstTrainingManualToEnd.java]: I don't know how

File: src/HomeWork3/FirstTrainingManualToEnd/Task11.java [hunk 24]
```diff
@@ -0,0 +1,20 @@
...
+         /*
+        * Task #11
+        *Имеется целое число, определить является ли это число простым, т.е.
+        делится без остатка только на 1 и себя.
+        * */
+        System.out.println("Task #11");
+
+        int naturalNumber = (int) (Math.random() * 45);
+
+        if ((naturalNumber % naturalNumber) == 0 && (naturalNumber % 1) == 0) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/Task11.java:14]: it is always true

File: src/HomeWork3/FirstTrainingManualToEnd/Task8.java [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+        /*
+        * Task #8
+        * Имеется целове число (задать с помощью Random rand = new Random(); int x =
+          rand.nextInt() ). Это число – количесво денег в рублях. Вывести это число, добавив к
+          нему слово «рублей» в правильном падеже.
+        * */
+        System.out.println("Task #8");
+        Random random = new Random();
+        int x = random.nextInt(28);
+        if (x == 1) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/Task8.java:16]: you can use x%10 for condition

File: src/HomeWork3/FirstTrainingManualToEnd/Task9.java [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+            Задание 9
+            Изменить пример с суммой чисел таким образом, чтобы рассчитывалась не сумма, а
+            произведение, т.е. факториал числа.
+        * */
+        System.out.println("Task #9");
+        double sumNumbers = 1;
+        int randomNumber = (int) (Math.random() * 20);
+        int count = 0;
+
+        while (randomNumber != 0) {
```
Comment (svetamolchanova) [inline src/HomeWork3/FirstTrainingManualToEnd/Task9.java:17]: factorial is 1*2*...*n

**Pull request closed without merging**